javascripts/events.js

```diff
@@ -1,13 +1,59 @@
 "use strict";
-
-const firebaseApi = require('./firebaseApi');
+const firebaseApi = require("./firebaseApi");
 const dom = require('./dom');
 console.log("dom in events", dom);
 
+// Object -> Boolean
+// Checks whether the object's "name" key includes the search text and returns true or false for filtering purposes
+const checkSearchMatch = (attraction) => {
+	let searchValue = new RegExp($("#searchBox").val().split(' ').join(`\s*`), "i");
+	console.log(searchValue);
+	return attraction.name.match(searchValue);
+};
+
+// Object -> Number
+// reduces an attraction object to just the value of its "area" key
+const getAttractionArea = (attraction) => {
+  return attraction.area_id;
+};
+
+// [Object] -> [Number]
+// takes in an array of attraction objects and returns an array of all of their area IDs
+const reduceToAreas = (attractionArray) => {
+  return attractionArray.map(getAttractionArea);
+};
+
+// removes the dotted border from any previous searches and then outlines any areas with attractions matching the given IDs
+const highlightAreas = (numArray) => {
+	while ($(".dotborder").length) {
```

Comment:
can you think of a better way to do this than a while loop?  if you had a lot of areas this wouldn't be very performant